buildSrc/src/main/java/lfformat/LfFormatStep.java

```diff
@@ -21,38 +23,75 @@ public final class LfFormatStep {
   private LfFormatStep() {}
 
   /** Return a {@code FormatterStep} for LF code. */
-  public static FormatterStep create(File projectRoot) {
+  public static FormatterStep create(File projectRoot) throws IOException {
     Step.projectRoot = projectRoot.toPath();
-    return new Step();
+    return Step.getInstance();
   }
 
   /** Implement LF-specific formatting functionality. */
-  public static class Step implements FormatterStep, Serializable {
+  private static class Step implements FormatterStep {
     // The use of the static keyword here is a workaround for serialization difficulties.
     /** The path to the lingua-franca repository. */
     private static Path projectRoot;
 
+    private static Step instance;
+
+    private static Process formatter;
+    private static Writer writer;
+
+    private static BufferedReader reader;
+    private static BufferedReader error;
```

Comment:
It is a little bit of a red flag that these variables are static. It is necessary because `Step` needs to be `Serializable`. I do not fully understand why this is required by Spotless, but I am under the impression that it relates to an optimization that we are not trying to use at the moment.